### PR TITLE
Added image upload functionality and changed name of index.html file

### PR DESCRIPTION
**Pull request by LilCSharp**
### What is a quick description of the change?

I added a JSP file and two Java servlets that have the ability to permit the retrieval of images uploaded to a test database

### Screenshot/Screengrab after this change?

image

### Is this fixing an issue?

I changed the name of the index.html file so that Alicia can use index.html for development

### Are there more details that are relevant?

none

### Any additional comments?

This must be completely refactored when integration with the front-end occurs.

### CODE REVIEW

**Review by wu-haotian — COMMENTED** [review]
I mentioned this in a previous PR as well... looks like there're multiple copies of the code, do you know why's that the case?

e.g. there are 4 imgHandling.js and they looks similar:
target/appengine-staging/js/imgHandling.js
target/step13-2020-1/js/imgHandling.js
target/appengine-staging/__static__/js/imgHandling.js
src/main/webapp/js/imgHandling.js

**Comment by LilCSharp** [comment]
These are not copies. These are files the mvn needs and automatically creates when running the dev server and when deploying the site. Everything with target after it is typically auto-generated

**Comment by wu-haotian** [comment]
> These are not copies. These are files the mvn needs and automatically creates when running the dev server and when deploying the site. Everything with target after it is typically auto-generated

I see, thanks for letting me know! If that's the case, I wonder if we should drop tracking those files with git. It makes code review more difficult, and may cause merge conflict in the future. @antrromet what's your suggestion here?

**Comment by jamierashley** [comment]
Code for the servlet looks good to me. You can delete the target folder before committing I believe, It's not necessary but it makes the code a little confusing.

**Comment by aliciasimon** [comment]
LGTM. The only thing I would say is to add some comments to your code so that people reviewing it can have a better understanding of what you've written.

File: src/main/java/com/google/sps/servlets/ServeImgServlet.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        }
+      }
+
+      if (key != null) {
+        BlobKey blobKey = new BlobKey(key);
+        blobstoreService.serve(blobKey, res);
+      }
+
+
+      /*
```
Comment (wu-haotian) [inline src/main/java/com/google/sps/servlets/ServeImgServlet.java]: nit: either remove commented-out code, or add a comment on when to put the code back (same for the commented-out codes in other files)

e.g. // TODO: un-comment the code block after XXX.

File: src/main/java/com/google/sps/servlets/ServeImgServlet.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        if (id == searchForID) {
+          break;
+        }
+      }
+
+      if (key != null) {
+        BlobKey blobKey = new BlobKey(key);
+        blobstoreService.serve(blobKey, res);
+      }
+
```
Comment (wu-haotian) [inline src/main/java/com/google/sps/servlets/ServeImgServlet.java]: nit: remove an extra empty line here

File: src/main/java/com/google/sps/servlets/ServeImgServlet.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import javax.servlet.annotation.WebServlet;
+
+@WebServlet("/serve")
+public class ServeImgServlet extends HttpServlet {
```
Comment (wu-haotian) [inline src/main/java/com/google/sps/servlets/ServeImgServlet.java:44]: add a javadoc to explain what does this class do.

File: src/main/java/com/google/sps/servlets/UploadImgServlet.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+@WebServlet("/upload")
+public class UploadImgServlet extends HttpServlet {
```
Comment (wu-haotian) [inline src/main/java/com/google/sps/servlets/UploadImgServlet.java:75]: add a javadoc to explain what does this class do.

File: src/main/java/com/google/sps/servlets/UploadImgServlet.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+             imgTest.setProperty("blobkey", blobKeys.get(0).getKeyString());
+             imgTest.setProperty("id", id);
+
+             datastore.put(imgTest);
+          }
+        }        
+        
+        res.sendRedirect("/Pages/index.jsp");
+
+        /*
```
Comment (wu-haotian) [inline src/main/java/com/google/sps/servlets/UploadImgServlet.java:140]: nit: remove or add comment for commented-out code blocks.

File: src/main/webapp/js/imgHandling.js [hunk 6]
```diff
@@ -44,6 +36,7 @@ function handleFiles() {
       display.onload = sketch;
       display.src = URL.createObjectURL(this.files[i]);
 
+      /*
```
Comment (wu-haotian) [inline src/main/webapp/js/imgHandling.js]: nit: remove or add comment for commented-out code blocks.

**Comment by wu-haotian** [comment]
I synced with Antriksh on the auto-generated code issue. Antriksh said that we'll need a gitignore file to do that. I've filed https://github.com/googleinterns/step13-2020/issues/17 for this, we can also discuss in team chat

It's okay to do that (drop tracking on auto-generated files/directories) in a follow up PR, but please still fix other issues I commented in previous code reviews.

**Comment by LilCSharp** [comment]
I added comments and documentation. By the time that the review was done, I also added some additional stuff to the file, but most of it is commented out, so there should not be much of a change in functionality

**Comment by wu-haotian** [comment]
> I added comments and documentation. By the time that the review was done, I also added some additional stuff to the file, but most of it is commented out, so there should not be much of a change in functionality

**Review by wu-haotian — APPROVED** [review]

**Pull request merged**